data/transactions/logic/eval_test.go

```diff
@@ -903,18 +903,11 @@ func TestArg(t *testing.T) {
 	t.Parallel()
 	for v := uint64(1); v <= AssemblerMaxVersion; v++ {
 		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
-			ops := testProg(t, `arg 0
-arg 1
-==
-arg 2
-arg 3
-!=
-&&
-arg 4
-len
-int 9
-<
-&&`, v)
+			source := "arg 0; arg 1; ==; arg 2; arg 3; !=; &&; arg 4; len; int 9; <; &&;"
+			if v >= 5 {
+				source += "int 0; args; int 1; args; ==; assert;"
```

Comment:
Could you also add a check that `int 2; args;` != `int 3; args;` too? These tests should protect against a bug where the opcode does not handle the index properly (e.g. ignores the stack arg and always assumes index 0 or something), and this helps prove the index is actually being used correctly.

Reply:
added in my followup